### PR TITLE
Create Repositories.md

### PR DESCRIPTION
**Pull request by nikithauc**
Creating a document to record the design and decisions for the repository set up.

### CODE REVIEW

File: Repositories.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+With the Kiota onboarding, our code will be using the following four solutions:
+- Kiota generator
+- Kiota core libraries
+- Graph core library which will be dependending on the Kiota core.
+- Graph service library containing Kiota generated the request builders.
```
Comment (SilasKenneth) [inline Repositories.md:5]:
```suggestion
- Graph service library containing Kiota generated request builders.
```

File: Repositories.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+With the Kiota onboarding, our code will be using the following four solutions:
+- Kiota generator
```
Comment (baywet) [inline Repositories.md:2]:
```suggestion

- Kiota generator
```

File: Repositories.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- Kiota core libraries
+- Graph core library which will be dependending on the Kiota core.
+- Graph service library containing Kiota generated the request builders.
+
+The proposed plan to set the GitHub repositories for the above solutions is as follows : 
+
+- Kiota (the generator) - common for all languages
+
```
Comment (baywet) [inline Repositories.md:10]:
```suggestion
```

File: Repositories.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- Kiota core libraries
+- Graph core library which will be dependending on the Kiota core.
+- Graph service library containing Kiota generated the request builders.
+
+The proposed plan to set the GitHub repositories for the above solutions is as follows : 
+
+- Kiota (the generator) - common for all languages
+	
+- kiota-<language>
+
```
Comment (baywet) [inline Repositories.md:12]:
```suggestion
```

File: Repositories.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- Graph service library containing Kiota generated the request builders.
+
+The proposed plan to set the GitHub repositories for the above solutions is as follows : 
+
+- Kiota (the generator) - common for all languages
+	
+- kiota-<language>
+	
+- msgraph-sdk-<language>-core
+
```
Comment (baywet) [inline Repositories.md:14]:
```suggestion
```

File: Repositories.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+The proposed plan to set the GitHub repositories for the above solutions is as follows : 
+
+- Kiota (the generator) - common for all languages
+	
+- kiota-<language>
+	
+- msgraph-sdk-<language>-core
+	
+- msgraph-sdk-<language>
+
```
Comment (baywet) [inline Repositories.md:16]:
```suggestion
```

File: Repositories.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+- Kiota (the generator) - common for all languages
+	
+- kiota-<language>
+	
+- msgraph-sdk-<language>-core
+	
+- msgraph-sdk-<language>
+	
+- msgraph-beta-sdk-<language>
+
```
Comment (baywet) [inline Repositories.md:18]:
```suggestion
```

File: Repositories.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+- kiota-<language>
+	
+- msgraph-sdk-<language>-core
+	
+- msgraph-sdk-<language>
+	
+- msgraph-beta-sdk-<language>
+	
+- (optional) msgraph-china-sdk-<language>
+
```
Comment (baywet) [inline Repositories.md:20]:
```suggestion
```

File: Repositories.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+	
+- msgraph-beta-sdk-<language>
+	
+- (optional) msgraph-china-sdk-<language>
+	
+- (optional) msgraph-beta-china-sdk-<language>
+ 
+Concerns to be discussed more:
+  
+1. Duplicate names - Consider adding new repositories for all the Kiota generated libraries and discontinuing the repos that use the current code generation process.
```
Comment (baywet) [inline Repositories.md:25]: in that case I suggest we rename the repo whenever we release to align with the other languages

File: Repositories.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+- Kiota generator
+- Kiota core libraries
+- Graph core library which will be dependending on the Kiota core.
+- Graph service library containing Kiota generated the request builders.
+
+The proposed plan to set the GitHub repositories for the above solutions is as follows : 
+
+- Kiota (the generator) - common for all languages
+	
+- kiota-<language>
```
Comment (baywet) [inline Repositories.md:11]:
```suggestion
- kiota-<language> : contains the abstractions and core implementation packages
```

File: Repositories.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+	
+- msgraph-sdk-<language>-core
+	
+- msgraph-sdk-<language>
+	
+- msgraph-beta-sdk-<language>
+	
+- (optional) msgraph-china-sdk-<language>
```
Comment (baywet) [inline Repositories.md:19]:
```suggestion
- (optional) msgraph-china-sdk-<language>: an advantage of Kiota based generation is we can deliver SDKs tailored to customers' environments, this includes the ability to ship one service library per national cloud per version as national clouds generally have access to fewer endpoints.
```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for putting all of this together! A couple of suggestions on the doc itself, and in addition to that, maybe we should add a link to that doc on one of the main pages to make it discoverable?

**Comment by MIchaelMainer** [comment]
For cloud specific SDKs that are not the public cloud, I suggest: msgraph{-beta}-{cloud name}-sdk-{language}.

With the Kiota onboarding, our code will be using the following four solutions:
 - Kiota generator
 - Kiota core libraries for each language - these are the common core HTTP client options for any client generated with *Kiota generator*.
 - Graph core library for each language which will be depending on the *Kiota core* libraries of the same language. These are the Graph specific implementations that provide Graph specific functionality.
-  Graph service library containing Kiota generated the request builders.

The name change game goes beyond repos. We need to consider whether the Kiota generated libraries and experience is a continuation of our existing libraries and experience, or is this an entirely new client library experience that requires us to divorce some or all of our existing experiences. We have accumulated many years worth of names, patterns, links, and other artifacts indexed across the Internet -- there is value in maintaining a relationship to those artifacts. There is value in a clean break. There will be confusion either way as we'd be introducing a change -- we need to think hard about how we mitigate that confusion. My default is to try to keep our existing names and links, and make sure we have a clear version upgrade that is well documented and discoverable.

**Comment by baywet** [comment]
@MIchaelMainer on the naming comment you made: I believe that for languages that already have a fluent story today, the kiota based version is just a new iteration of that and would should reuse the names (dotnet, java,...). For other languages that don't have a fluent API story today, we should align, either by creating the new libs/repos if inexistent, or by renaming the existing repos if they are in the way of being aligned with other languages.

File: Repositories.md (hunk 9 above)
Comment (baywet) [reply Repositories.md:20]:
```suggestion
1. Duplicate names - Consider adding new repositories for all the Kiota generated libraries and discontinuing the repos that use the current code generation process. Also consider renaming the older repository.
```

**Pull request merged**